src/core/src/sim/mc/moves/translate.rs

```diff
@@ -20,34 +20,33 @@ pub struct Translate {
     molid: usize,
     /// New positions of the atom in the translated molecule
     newpos: Vec<Vector3D>,
-    /// Translation vector
-    delta: Vector3D,
+    /// Maximum displacement value
+    delta: f64,
     /// Translation range for random number generation
     delta_range: Range<f64>,
 }
 
 impl Translate {
-    /// Create a new `Translate` move, with maximum displacement of `dr`,
-    /// translating all the molecules in the system.
+    /// Create a new `Translate` move, with maximum displacement of `dr`.
+    /// Translating all the molecules in the system.
     pub fn new(dr: f64) -> Translate {
         Translate::create(dr, None)
     }
 
-    /// Create a new `Translate` move, with maximum displacement of `dr`,
-    /// translating only molecules with `moltype` type.
+    /// Create a new `Translate` move, with maximum displacement of `dr`.
+    /// Translating only molecules with `moltype` type.
     pub fn with_moltype(dr: f64, moltype: u64) -> Translate {
         Translate::create(dr, Some(moltype))
     }
 
     /// Factorizing the constructors
     fn create(dr: f64, moltype: Option<u64>) -> Translate {
         assert!(dr > 0.0, "dr must be positive in Translate move");
-        let dr = dr / f64::sqrt(3.0);
```

Comment:
I think you need to keep this division by sqrt(3) to ensure that the vector norm will be lower than the given dr.